### PR TITLE
PIN-3673 Include parent organization name in AOO and UO name

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 1]
```diff
@@ -128,8 +128,12 @@ object IPAOpenDataServiceImpl {
   private def getDescription(institutionKind: InstitutionKind, mapped: Map[String, Int], record: List[RecordValue]) =
     institutionKind match {
       case InstitutionKind.Agency => record(mapped(InstitutionsFields.agencyDescription)).select[String]
-      case InstitutionKind.AOO    => record(mapped(InstitutionsFields.aooDescription)).select[String]
-      case InstitutionKind.UO     => record(mapped(InstitutionsFields.uoDescription)).select[String]
+      case InstitutionKind.AOO    =>
+        Some(s"${record(mapped(InstitutionsFields.aooDescription))
+            .select[String]} - ${record(mapped(InstitutionsFields.agencyDescription)).select[String]}")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]: `.select[]` returns and `Option`. This string concatenation would result in the string
`Some(AOO name) - Some(Institution Name)`

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 2]
```diff
@@ -128,8 +128,12 @@ object IPAOpenDataServiceImpl {
...
+      case InstitutionKind.UO     =>
+        Some(s"${record(mapped(InstitutionsFields.uoDescription))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]: same

**Review by galales — CHANGES_REQUESTED** [review]

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 3]
```diff
@@ -128,8 +128,14 @@ object IPAOpenDataServiceImpl {
   private def getDescription(institutionKind: InstitutionKind, mapped: Map[String, Int], record: List[RecordValue]) =
     institutionKind match {
       case InstitutionKind.Agency => record(mapped(InstitutionsFields.agencyDescription)).select[String]
-      case InstitutionKind.AOO    => record(mapped(InstitutionsFields.aooDescription)).select[String]
-      case InstitutionKind.UO     => record(mapped(InstitutionsFields.uoDescription)).select[String]
+      case InstitutionKind.AOO    =>
+        Some(s"${record(mapped(InstitutionsFields.aooDescription))
+            .select[String]
+            .getOrElse("")} - ${record(mapped(InstitutionsFields.agencyDescription)).select[String].getOrElse("")}")
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]:
```suggestion
      case InstitutionKind.AOO    =>
        for {
          aoo    <- record(mapped(InstitutionsFields.aooDescription)).select[String]
          agency <- record(mapped(InstitutionsFields.agencyDescription)).select[String]
        } yield s"$aoo – $agency"
```

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 4]
```diff
@@ -128,8 +128,14 @@ object IPAOpenDataServiceImpl {
...
+      case InstitutionKind.UO     =>
+        Some(s"${record(mapped(InstitutionsFields.uoDescription))
+            .select[String]
+            .getOrElse("")} - ${record(mapped(InstitutionsFields.agencyDescription)).select[String].getOrElse("")}")
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]:
```suggestion
      case InstitutionKind.UO     =>
        for {
          uo     <- record(mapped(InstitutionsFields.uoDescription)).select[String]
          agency <- record(mapped(InstitutionsFields.agencyDescription)).select[String]
        } yield s"$uo – $agency"
```

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/util/DocumentConverter.scala [hunk 5]
```diff
@@ -19,7 +22,12 @@ object DocumentConverter {
         originId = document.get(InstitutionField.ORIGIN_ID.value),
         taxCode = document.get(InstitutionField.TAX_CODE.value),
         category = document.get(InstitutionField.CATEGORY.value),
-        description = document.get(InstitutionField.DESCRIPTION.value),
+        description = classification match {
+          case AGENCY => document.get(InstitutionField.DESCRIPTION.value)
+          case AOO    =>
+            s"${document.get(InstitutionField.AOO.value)} - ${document.get(InstitutionField.DESCRIPTION.value)}"
+          case UO => s"${document.get(InstitutionField.OU.value)} - ${document.get(InstitutionField.DESCRIPTION.value)}"
+        },
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/util/DocumentConverter.scala]:
```suggestion
       description = document.get(InstitutionField.DESCRIPTION.value),
```

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**